libkbfs/folder_block_ops.go

```diff
@@ -1347,6 +1347,17 @@ func (fbo *folderBlockOps) writeDataLocked(
 			newb := fblock.IPtrs[len(fblock.IPtrs)-1]
 			copy(fblock.IPtrs[indexInParent+2:], fblock.IPtrs[indexInParent+1:])
 			fblock.IPtrs[indexInParent+1] = newb
+			if oldSize == de.Size {
+				// For the purposes of calculating the newly-dirtied
+				// bytes for the deferral calculation, disregard the
+				// existing "hole" in the file.
+				oldSize = uint64(newb.Off)
+			}
+		}
+
+		// Nothing was copied, no need to dirty anything.
+		if nCopied == oldNCopied {
```

Comment:
Yes.  It happens when the block is already full.  Specifically it can happen with the truncate case like this, after a `truncate 777777` with 65KB blocks:

Indirect pointer offsets: `[0,66559],<HOLE>,[777777]`
New write: `write off=66560`

In this case `getFileBlockAtOffsetLocked` returns block `[0,66559]`, which is already full, and so 0 new bytes are copied (it then makes a new block).  I guess it can also happen even without holes, if the final block is completely full when a new append comes in.

I can add a (briefer) comment explaining.